testing/tests/DevExpress.viz.funnel/funnel.label.tests.js

```diff
@@ -1300,3 +1301,57 @@ QUnit.test('change resolveLabelOverlapping option', function(assert) {
     const label = labelModule.Label.getCall(0).returnValue;
     assert.ok(label.shift.called);
 });
+
+QUnit.module('Initialization', {
+    beforeEach: function() {
+        this.loggerErrorSpy = sinon.spy(logger, 'error');
+    },
+    afterEach: function() {
+        this.loggerErrorSpy.restore();
+    }
+}, () => {
+    [true, false].forEach((showForZeroValues) => {
+        QUnit.test(`should not fail when showForZeroValues==${showForZeroValues} and datasource contains zero values (T1172293)`, function(assert) {
+            createFunnel({
+                algorithm: 'dynamicHeight',
+                showForZeroValues,
+                dataSource: [
+                    { count: 1, level: 'Junior Engineer' },
+                    { count: 20, level: 'Mid-Level Engineer' },
+                    { count: 0, level: 'Senior Engineer' },
+                    { count: 17, level: 'Architect' },
+                ],
+                valueField: 'count',
+                inverted: true,
+                sortData: false,
+            });
+
+            assert.strictEqual(this.loggerErrorSpy.callCount, 0);
```

Comment:
The logger is not useful here. Widget failed and this assert will not work. The best way to check is wrap the createFunnel in try catch like in [datebox](https://github.com/DevExpress/DevExtreme/blob/1ce4576a12c62e36f6eb714f0df8bf9a5f2e6807/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js#L228) tests. When the test is failed, there will be a full message what is wrong